pimd/pim_bsm.c

```diff
@@ -620,9 +623,14 @@ void pim_bsm_clear(struct pim_instance *pim)
 		}
 
 		/* Deregister addr with Zebra NHT */
-		nht_p.family = AF_INET;
-		nht_p.prefixlen = IPV4_MAX_BITLEN;
+		nht_p.family = PIM_AF;
+		nht_p.prefixlen = PIM_MAX_BITLEN;
+#if PIM_IPV == 4 || !defined(PIM_V6_TEMP_BREAK)
 		nht_p.u.prefix4 = rp_info->rp.rpf_addr.u.prefix4;
```

Comment:
We should just use `nht_p = rp_info->rp.rpf_addr` here, replace 8 lines with 1.

Reply:
Done